src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_GROW_INDEX = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size + 1 > threshold) {
+            resizeTable();
+        }
+        Node<K, V> current = new Node<>(key, value, null);
+        int index;
+        if (key == null) {
+            index = 0;
+        } else {
+            index = Math.abs(getHash(key) % table.length);
```

Comment:
As I can see u are using the same code many times. In my opinion, it will be better if u will create a dedicated method to compute your index

Reply:
`index = Math.abs(getHash(key) % table.length);`